addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,597 @@
+-------------------------------------------------------------------------------
+function init_granules()
+    Granules = T{"Crimson granules of time","Azure granules of time","Amber granules of time",
+                "Alabaster granules of time","Obsidian granules of time"}
+    for i=1, #Granules do
+        w[Granules[i]] = 0
+    end
+end
+init_granules()
+
+-------------------------------------------------------------------------------
+-- The on screen window structure ---------------------------------------------
+-------------------------------------------------------------------------------
+function init_window()
+    local showCurrencyDivider = false
+    window.text(window,Yellow.."Time remaining: ${time|initializing...}")
+    window.appendline(window,"\\cr————————————————————")
+    window.appendline(window,"${current_mob|(unknown)}\n"..Green.."${current_proc|(none)}")
+    window.appendline(window,"\\cr————————————————————")
+    for i=1, #Currency do
+        if w[Currency[i]] > 0 then
+            showCurrencyDivider = true
+            if Currency[i] == "Ordelle Bronzepiece" or Currency[i] == "Montiont Silverpiece" then
+                window.appendline(window,Currency[i]..": ${"..Currency[i].."|0}")
+            elseif Currency[i] == "One Byne Bill" or Currency[i] == "One Hundred Byne Bill" then
+                window.appendline(window,Currency[i]..": ${"..Currency[i].."|0}")
+            elseif Currency[i] == "Tukuku Whiteshell" or Currency[i] == "Lungo-Nango Jadeshell" then
+                window.appendline(window,Currency[i]..": ${"..Currency[i].."|0}")
+            else
+                window.appendline(window,"\\cr"..Currency[i]..": ${"..Currency[i].."|0}")
+            end
+        end
+    end
+    if showCurrencyDivider then
+        window.appendline(window,"\\cr————————————————————")
+    end
+    for i=1, #Granules do
+        if(w[Granules[i]] == 1) then
+            window.appendline(window,Green..Granules[i])
+        else
+            window.appendline(window,Red..Granules[i])
+        end
+    end
+end
+init_window()
+
+-------------------------------------------------------------------------------
+-- Register a prerendere event for the display refresh ------------------------
+-------------------------------------------------------------------------------
+windower.register_event('prerender', function()
+    if obj_time < 1 then return end
+    if obj_time ~= (end_time - os.time()) then
+        obj_time = end_time - os.time()
+        w.time = os.date('!%H:%M:%S', obj_time)
+        window:update(w)
+    end
+end)
+-------------------------------------------------------------------------------
+-- Did we enter a Dynamis Zone? -----------------------------------------------
+-------------------------------------------------------------------------------
+windower.register_event('zone change', function(zone)
+    if ProcZones:contains(windower.ffxi.get_info().zone) then
```

Comment:
No need to call `windower.ffxi.get_info` here, just to get the zone; it's already a function argument.